### DISCUSSION TITLE
bind cannot return an array

### DISCUSSION THREAD

**Discussion by FredericHeem** [vanjs-org/van#43] [General]
Is there any reason why bind cannot return an array of dom element ?  If the array in wrapped into a div, the consequence is to re-render all elements in the array,  which is in some use cases a problem, especially when the item is animated.

**Comment by Tao-VanJS** [comment]
Handling reactive state binding for an array of DOM elements requires much more complex handling compared to the current implementation (a diffing algorithm is likely needed). It's at odds with the minimalist design of VanJS project. If the performance of rendering all elements in an array is a major performance concern for your project, you can consider the feature [stateful binding](https://vanjs.org/tutorial#stateful-binding), which allows you to implement customized optimization to avoid over-rendering.
